src/components/calcite-input/calcite-input.tsx

```diff
@@ -484,7 +483,7 @@ export class CalciteInput {
     const decimalSeparator = getDecimalSeparator(this.locale);
     if (
       event.key === decimalSeparator &&
-      isValidDecimal(this.step === "any" ? 1 : (this.step as number))
+      (this.step === "any" ? true : isValidDecimal(this.step))
```

Comment:
Nitpick: This can be simplified to `this.step === "any" || isValidDecimal(this.step)`

Reply:
Yeah for some reason I had it that way and it was allowing decimals in integer-only mode, but now that I add it back in I'm seeing it work.  Must have been a dev server problem where the changes weren't loading right or something.